contracts/ServiceRegistryTokenUtility.sol

```diff
@@ -254,15 +373,19 @@ contract ServiceRegistryTokenUtility is IErrorsRegistries {
             if (allowance < securityDeposit) {
                 revert IncorrectRegistrationDepositValue(allowance, securityDeposit, serviceId);
             }
+            // Set the token-secured flag for the service
+            isTokenSecured = true;
 
             // Transfer tokens from the serviceOwner account
-            // TODO Re-entrancy
-            // TODO Safe transferFrom
-            bool success = IToken(token).transferFrom(serviceOwner, address(this), securityDeposit);
-            if (!success) {
-                revert TransferFailed(token, serviceOwner, address(this), securityDeposit);
+            uint256 balanceBefore = IToken(token).balanceOf(address(this));
+            safeTransferFrom(token, serviceOwner, address(this), securityDeposit);
+            uint256 balanceAfter = IToken(token).balanceOf(address(this));
+            // Check the correctness of received funds
+            if (balanceBefore > balanceAfter || (balanceAfter - balanceBefore) != securityDeposit) {
```

Comment:
bigger or equal?

Reply:
bigger for avoid x - y < 0

Reply:
First check is the reverse one - if before is bigger than after, meaning that during the transfer the contract has lost funds :) as if the token is really bad. In this case we can't get the diff, safe math would fail going underflow.

Reply:
Yeah, instead equal to zero is handled by the second operator